### PR TITLE
circleci/v2: add CancelWorflow function

### PR DESCRIPTION
**Pull request by cpanato**
#### Summary
Also this is needed and good, to start slowing moving to circleci API v2

circleci/v2: add CancelWorflow function

#### Ticket Link
Part of https://mattermost.atlassian.net/browse/DOPS-430

### CODE REVIEW

File: circleci.go [hunk 1]
```diff
@@ -1072,6 +1072,32 @@ func (c *Client) GetPipelineByBranchWithContext(ctx context.Context, vcsType Vcs
 	return p, nil
 }
 
+// CancelBuild triggers a cancel of the specified workflow using CirclerCI apiV2
```
Comment (ethervoid) [inline circleci.go]:
```suggestion
// CancelWorkflow triggers a cancel of the specified workflow using CirclerCI apiV2
```

File: circleci.go [hunk 2]
```diff
@@ -1072,6 +1072,32 @@ func (c *Client) GetPipelineByBranchWithContext(ctx context.Context, vcsType Vcs
...
+
+// CancelWorkflowWithContext is the same as CancelWorkflow with the addition of the context
+// parameter that would be used to request cancellation.
+func (c *Client) CancelWorkflowWithContext(ctx context.Context, workflowID string) (*CancelWorkflow, error) {
+	if c.Version < APIVersion2 {
+		return nil, newInvalidVersionError(c.Version)
+	}
+	cancel := &CancelWorkflow{}
+
+	err := c.request(ctx, "POST", fmt.Sprintf("workflow/%s/cancel", workflowID), cancel, nil, nil)
```
Comment (ethervoid) [inline circleci.go]:
```suggestion
	err := c.request(ctx, http.MethodPost, fmt.Sprintf("workflow/%s/cancel", workflowID), cancel, nil, nil)
```

**Review by ethervoid — APPROVED** [review]
LGTM! Some small nits

File: circleci.go (hunk 1 above)
Comment (cpanato) [reply circleci.go]: again :/

File: circleci.go (hunk 2 above)
Comment (cpanato) [reply circleci.go]: done, thanks for the review

**Review by ethervoid — APPROVED** [review]
LGTM

**Review by agnivade — APPROVED** [review]

**Pull request merged**